db/migrations/postgres/000037_create_plugin_key_value_store.up.sql

```diff
@@ -0,0 +1,20 @@
+CREATE TABLE IF NOT EXISTS pluginkeyvaluestore (
+    pluginid VARCHAR(190) NOT NULL,
+    pkey VARCHAR(50) NOT NULL,
+    pvalue bytea,
+    PRIMARY KEY (pluginid, pkey)
+);
+
+ALTER TABLE pluginkeyvaluestore ADD COLUMN IF NOT EXISTS expireat bigint DEFAULT 0;
+
+DO $$BEGIN
+    IF (
+        SELECT column_default::bigint
+        FROM information_schema.columns
+        WHERE table_schema='public'
+        AND table_name='pluginkeyvaluestore'
+        AND column_name='expireat'
+    ) = 0 THEN
+        ALTER TABLE pluginkeyvaluestore ALTER COLUMN expireat SET DEFAULT NULL;
```

Comment:
Looks to me that the `ExpireAt` column was originally created with a default (version 5.6) and then it was removed (version 5.8). Here we are combining the two changes in the same migration file so I am not sure it makes sense to set the default on column creation.

But the overall question from my side is: do we have a consistent way of mapping these migration files to MM schema versions? Here we are merging changes from multiple versions.

cc @isacikgoz

Reply:
The idea of combining the two changes in idempotent steps is that it doesn't matter in which version you are and what's the associated status of your schema, if you apply the migration you'll always end up in the last state. That's why we go through "all the steps" even if we know that we will undo them right after.

> do we have a consistent way of mapping these migration files to MM schema versions?

Nope, we don't. Based on the principle above, the idea is to run all this migrations for all instances and get to a valid "versionless" starting point, and move forward from there.

Reply:
I wonder what if a migration step requires other migration to complete? So far I haven't checked that possibility.

Reply:
Okay, does this mean that we are dropping previous schema versions? E.g. As soon as this new system runs we cannot go back to some previous version (as in 5.x). But afterwards I'd imagine that a file (or a group of these files) will define a version (as we think of it today).

Reply:
> I wonder what if a migration step requires other migration to complete? So far I haven't checked that possibility.

that'd be caught by tests, so I think that so far we're good :) The way to stablish dependencies now would be just to tweak the sequence / split a migration. We planned for a declaring required migrations feature in `morph`, but it's still to get implemented

> does this mean that we are dropping previous schema versions?

Probably not yet just in case, but once we have the new migration system running, I'd say yes, we should add a migration to delete the schemas table and deprecate the code around it at some point

> But afterwards I'd imagine that a file (or a group of these files) will define a version (as we think of it today)

Exactly!